presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1134,4 +1148,10 @@ public static boolean isOptimizedRepartitioningEnabled(Session session)
     {
         return session.getSystemProperty(OPTIMIZED_REPARTITIONING_ENABLED, Boolean.class);
     }
+
+    public static boolean isExactPartitioningPreferred(Session session)
```

Comment:
I did it this way to simplify usage because we only have a binary behavior based on this flag right now. If I return the enum value then I have to put the check against the enum value. It's internal so we can always refactor it to return the enum later when we do something with it.

isXYX is done 37 times in SystemSessionProperties so I would argue that this is consistent. :-P

I'm going to mark this as resolved unless you want to discuss it more.